package.json

```
@@ -36,6 +36,16 @@
 					],
 					"default": 3000,
 					"description": "Specifies the Git branch polling interval in milliseconds."
+				},
+				"branchInWindowTitle.branchTemplate": {
+					"type": "string",
+					"default": "${separator}[Branch: $branchName]",
```

Rather than having another setting for prefix/suffix, what do you think about something like this?

```suggestion
					"default": "${originalTitle} ${separator} [Branch: ${branchName}]",
```

Where `${originalTitle}` is the original title from VS Code before we alter it.

Can we also keep the previous behaviour as default? I.e. have spaces before and after the separator.

---

Do NOT add spaces manually. `${separator}` already includes them.
What to the `${originalTitle}`, the code become much harder to understand. That was my initial approach once I discovered the instability of the modifications application.

---

Can you explain the use case that is motivating this PR? Do you need to be able to customise _both_ the rendering of the branch text _and_ its placement (suffix/prefix)?

---

For positioning: Without calling names, I have a time tracking application that assumes that everything after last dash is an application name. Seeing "application names" like "655" for Jira-style branch names (i.e. "DEVSTR-655") is not very useful when you want to categorize your activities.

For customizability: I prefer concise presentations at all times. So that now my title looks like
`[<branch>] <project>:<medium path name> - <App name>`
Nice and clear.